src/main/java/org/jenkinsci/plugins/p4/groovy/P4Groovy.java

```diff
@@ -14,19 +14,22 @@
 import java.util.Arrays;
 import java.util.List;
 import java.util.Map;
+import org.jenkinsci.plugins.p4.credentials.P4BaseCredentials;
 
 public class P4Groovy implements Serializable {
 
 	private static final long serialVersionUID = 1L;
 
-	private final String credential;
+	private final String credentialName;
 	private final Workspace workspace;
 	private final FilePath buildWorkspace;
+	private final P4BaseCredentials p4Credential;
```

Comment:
I need to fully understand this change, but having both the credential ID and the class seems wrong.  It should be possible to look up the credential.